doc/api/quic.md

```diff
@@ -0,0 +1,1640 @@
+
+session.datagram('hello world datagram');
+
+session.ondatagram = ({ datagram }) => {
+  console.log(dec.decode(datagram));
+};
+```
+
+```cjs
+const {
+  Endpoint,
+} = require('net/quic');
+
+async function sayHi() {
+  const client = new Endpoint();
+  const session = client.connect({
+    address: '123.123.123.123',
+    port: 1234,
+  }, { alpn: 'xyz' });
+
+  await session.handshake;
+
+  session.datagram('hello world datagram');
+
+  session.ondatagram = ({ datagram }) => {
+    console.log(dec.decode(datagram));
+  };
+}
+
+sayHi();
+```
+
+### HTTP/3
+
+Support for the HTTP/3 application protocol is built in to the QUIC
+implementation. Using HTTP/3 with QUIC uses the same basic API with
+some additional bits for headers and trailers.
+
+#### A simple HTTP/3 Server
+
+```mjs
+import {
+  Endpoint,
+} from 'node:net/quic';
+
+import { stdout } from 'node:process';
+
+const endpoint = new Endpoint({ address: { port: 12345 } });
+
+endpoint.onsession = ({ session }) => {
+  session.onstream = async ({ stream, respondWith }) => {
+    respondWith({
+      headers: { 'content-type': 'text/plain' },
+      body: 'right back at you',
+    });
+
+    console.log(await stream.headers);
+
+    stream.readableNodeStream().pipe(stdout);
+  };
+};
+
+endpoint.listen({
+  secure: {
+    key: getTLSKeySomehow(),
+    cert: getTLSCertSomehow(),
+  },
+});
+```
+
+```cjs
+const {
+  Endpoint,
+} = require('net/quic');
+
+const { stdout } = require('process');
+
+const endpoint = new Endpoint({ address: { port: 12345 } });
+
+endpoint.onsession = ({ session }) => {
+  session.onstream = async ({ stream, respondWith }) => {
+    respondWith({
+      headers: { 'content-type': 'text/plain' },
+      body: 'right back at you',
+    });
+
+    console.log(await stream.headers);
+
+    stream.readableNodeStream().pipe(stdout);
+  };
+};
+
+endpoint.listen({
+  secure: {
+    key: getTLSKeySomehow(),
+    cert: getTLSCertSomehow(),
+  },
+});
+```
+
+#### A simple HTTP/3 Client
+
+```mjs
+import {
+  Endpoint,
+} from 'node:net/quic';
+
+import { stdout } from 'node:process';
+
+const client = new Endpoint();
+
+const req = client.connect({
+  address: '127.0.0.1',
+  port: 12345
+}, {
+  hostname: 'localhost',
+});
+
+await req.handshake;
+
+const stream = req.open({
+  headers: { ':method': 'POST' },
+  body: 'hello there',
+});
+
+console.log(await stream.headers);
+
+stream.readableNodeStream().pipe(stdout);
+
+await client.close();
+```
+
+```cjs
+const {
+  Endpoint,
+} = require('net/quic');
+
+const { stdout } = require('process');
+
+const client = new Endpoint();
+
+const req = client.connect({
+  address: '127.0.0.1',
+  port: 12345
+}, {
+  hostname: 'localhost',
+});
+
+(async () => {
+  await req.handshake;
+
+  const stream = req.open({
+    headers: { ':method': 'POST' },
+    body: 'hello there',
+  });
+
+  console.log(await stream.headers);
+
+  stream.readableNodeStream().pipe(stdout);
+
+  await client.close();
+})();
```

Comment:
```suggestion
req.handshake.then(async () => {
  const stream = req.open({
    headers: { ':method': 'POST' },
    body: 'hello there',
  });

  console.log(await stream.headers);

  stream.readableNodeStream().pipe(stdout);

  await client.close();
});
```
?

Reply:
I prefer the async/await syntax version.